src/coreclr/vm/arm64/asmhelpers.asm

```diff
@@ -473,15 +473,14 @@ ShadowUpdateDisabled
 CheckCardTable
         ; Branch to Exit if the reference is not in the Gen0 heap
         ;
-        adr      x12,  wbs_ephemeral_low
-        ldp      x12,  x16, [x12]
+        ldr      x12,  wbs_ephemeral_low
         cbz      x12,  SkipEphemeralCheck
-
         cmp      x15,  x12
         blo      Exit
 
-        cmp      x15,  x16
-        bhi      Exit
+        ldr      x12,  wbs_ephemeral_high
+        cmp      x15,  x12
+        bhs      Exit
```

Comment:
The `.S` version is using `ccmp` here. I think we should use the same pattern in both cases. 

I wonder which would be better though. In theory the ephemeral check should more often fail than pass, statistically. In a large heap most objects are tenured. Ephemeral set is supposed to be small.

On the other hand both patterns probably work the same on a speculative CPU, but the predicated pattern is shorter by one instruction.

Reply:
And on server GC ephemeral check will always pass. I think, since we are touching this, it should be switched to the predicated form like in `.S`